src/workspace/discovery.rs

```diff
@@ -701,10 +701,111 @@ fn resolve_workspace_for_config_folder(
         }
       })
     };
+
+  enum ConfigFileName {
+    Single(&'static str),
+    Multiple(&'static [&'static str]),
+  }
+
+  let collect_member_config_folders = |kind: &'static str,
+                                       pattern_members: Vec<&String>,
+                                       file_specifier: &Url,
+                                       dir_path: &Path,
+                                       config_file_name: ConfigFileName|
+   -> Result<
+    Vec<PathBuf>,
+    WorkspaceDiscoverErrorKind,
+  > {
+    let patterns = pattern_members
+      .iter()
+      .map(|raw_member| {
+        let names = match config_file_name {
+          ConfigFileName::Single(name) => vec![name],
+          ConfigFileName::Multiple(names) => names.to_vec(),
+        };
```

Comment:
```suggestion
        let names = match config_file_name {
          ConfigFileName::Single(name) => vec![name],
          ConfigFileName::Multiple(names) => names.to_vec(),
        };
```

Maybe just always pass `&'static [&'static str]` here and `.iter()` on it to avoid allocating an intermediate vector?